ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -225,10 +213,9 @@ public void handlePost(RestRequest restRequest, RestResponseChannel restResponse
     long processingStartTime = System.currentTimeMillis();
     long preProcessingTime = 0;
     handlePrechecks(restRequest, restResponseChannel);
-    restRequest.getMetricsTracker().injectMetrics(frontendMetrics.postBlobMetrics);
-    RestRequestMetrics requestMetrics =
-        restRequest.getSSLSession() != null ? frontendMetrics.postBlobSSLMetrics : frontendMetrics.postBlobMetrics;
-    restRequest.getMetricsTracker().injectMetrics(requestMetrics);
+    boolean sslUsed = restRequest.getSSLSession() != null;
+    RestRequestMetrics metrics = frontendMetrics.postRequestMetricsGroup.getRestRequestMetrics(sslUsed, false);
+    restRequest.getMetricsTracker().injectMetrics(metrics);
```

Comment:
So we set up metrics up here and maybe overwrite it later, as opposed to just setting it up on line 232 so that the restRequest has a metrics instance when `securityService.preProcessRequest()` is called?

Reply:
yes. you are right.